core/modules/system/system.install

```diff
@@ -664,6 +786,23 @@ function system_requirements($phase) {
   return $requirements;
 }
 
+/**
+ * Gets the size of a specified directory.
+ *
+ * @param string $directory
+ *
+ * @return integer
+ */
+function _system_get_dir_size($directory) {
+  $size = 0;
+
+  foreach (new RecursiveIteratorIterator(new RecursiveDirectoryIterator($directory)) as $file) {
```

Comment:
This means, on *every* call of the status page, we do a full iteration, which can be an expensive task with many files in public/private.

Reply:
Yeah. ...wonder if caching the results makes sense in this scenario 🤔 ...on the other hand, how often would an admin visit that page? It's not like they'd sit there refreshing the page all day, right?

Reply:
My concern is that the status page is already sort of "expensive" - many checks are done in one request. Now think of a site with thousands of files... it might even be that the status page runs into a timeout (or PHP out of resources, worst case).